src/vs/workbench/services/extensions/common/extensions.ts

```diff
@@ -222,6 +222,9 @@ export interface IExtensionService extends ICpuProfilerTarget {
 	 * Stops the extension host.
 	 */
 	stopExtensionHost(): void;
+
+	getNamedCustomer?(sid: string): any;
```

Comment:
not sure about these changes?

Reply:
Here's how I wired things up. The `CodeInsetController` needs to create a `WebviewInputEditor` in order to pass it to `resolveCodeInset`. This `WebviewInputEditor` needs to be one for which a proxy exists, so that an extension can call methods on it. This is the job of `MainThreadWebviews`. So, to connect the dots between `CodeInsetController` and `MainThreadWebviews`, I added this new method `getNamedCustomer` on `IExtensionService` to allow the following code:

```
	private getWebviewService(): MainThreadWebviews {
		if (!this._mainThreadWebviews) {
			this._mainThreadWebviews = this._extensionService.getNamedCustomer('MainThreadWebviews');
		}
		return this._mainThreadWebviews;
	}
```

Sorry if I'm violating architectural intentions. I'm a n00b. :)

Reply:
Looks a little fishy but I also have to think of a better way first....

Reply:
Remove the named customer thing because that was the wrong kind of layering. It's now the webview main thread object reaching into the code inset conrtoller. also not the nicest but now layer breaker and it will do its job until we have settled on the APIs